### DISCUSSION TITLE
Resizing Linux distribution disk layouts

### DISCUSSION THREAD

**Discussion by rismoney** [vmware/packer-examples-for-vsphere#267] [Help] [answered]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

I had a queation about the partition scheme used.  I get that it is based on CIS standards of partition separation.   What is unclear to me, is how to manage it.  On this repo as is, you are using 60GB drives, but only 38GB seem being used.  If using terraform to then deploy to the 60GB or to 100GB, is there any guidance on how to grow a partition, ie say /var through the automation?  Is the 22GB unallocated intentional?   I am new to this, so if I am off base let me know.  Thanks, and awesome work on this effort!!

**Comment by tenthirtyam** [comment]
Transferred to Discussions.

**Comment by rismoney** [comment]
I didn't realize discussions was available :)

This is what I sort of came to understand for this image.
Am I on the right track?

growpart /dev/sda 3
lvextend -L +2G /dev/mapper/sysvg-var
xfs_growfs /dev/mapper/sysvg-var

**Comment by tenthirtyam** [comment, marked as answer]
When using the Terraform provider for vSphere, for example, you could use the `remote-exec` provisioner to run inline commands or an array scripts to perform the task.

```hcl
resource "null_resource" "extend_storage" {

  provisioner "remote-exec" {
    inline = [
      "sudo ... ",
    ]
    connection {
      type     = "ssh"
      host     = vsphere_virtual_machine.my_machine.vapp[0].properties.ip0
      user     = var.my_username
      password = var.my_password
      port     = "22"
      agent    = false
    }
  }
}

```
